puppet/lib/puppet/util/puppetdb/http.rb

```diff
@@ -141,7 +168,14 @@ def self.failover_action(path_suffix, server_urls, sticky, ssl_context = nil, ht
           end
 
           response = Timeout.timeout(config.server_url_timeout) do
-            http_callback.call(http, route)
+            response = http_callback.call(http, route)
+
+            if config.submit_only_server_urls.include?(server_url)
+              # Clear any previously registered timeout for this URL
+              @@submit_only_url_timeouts[server_url] = false
```

Comment:
I believe we will need to copy this logic in `broadcast_action` as well. It is configurable in the `action` method below which one a command will use.

Reply:
Okay, yeah found a testing hole now that you point this out, also. Working to fix that up.